frontend/app/controllers/Joiner.scala

```diff
@@ -259,14 +259,17 @@ object Joiner extends Controller with ActivityTracking
         // errors due to user's card are logged at WARN level as they are not logic errors
         case error: Stripe.Error =>
           logger.warn(s"Stripe API call returned error: \n\t$error \n\tuser=${request.user.id}")
+          setBehaviourNote(tier.name, error.code)(request)
           Forbidden(Json.toJson(error))
 
         case error: PaymentGatewayError =>
+          setBehaviourNote(tier.name, error.code)(request)
           handlePaymentGatewayError(error, request.user.id, tier.name, idRequest.trackingParameters, formData.deliveryAddress.countryName)
 
         case error =>
           salesforceService.metrics.putFailSignUp(tier)
           logger.error(s"User ${request.user.id} could not become ${tier.name} member: ${idRequest.trackingParameters}", error)
+          setBehaviourNote(tier.name, "card_error")(request)
```

Comment:
This could also catch unknown PayPal errors when PayPal goes live. The name `card_error` might make people assume there's a problem with Stripe, so this may either need to be more generic, or we may need to split out the errors more when the time comes. It'll probably be trial and error catching specific PayPal errors because the docs are limited here... 😞